core/src/main/scala/org/broadinstitute/dsde/rawls/entities/datarepo/DataRepoBigQuerySupport.scala

```diff
@@ -0,0 +1,145 @@
+package org.broadinstitute.dsde.rawls.entities.datarepo
+
+import com.google.cloud.bigquery.Field.Mode
+import com.google.cloud.bigquery._
+import org.broadinstitute.dsde.rawls.entities.exceptions.{DataEntityException, EntityNotFoundException}
+import org.broadinstitute.dsde.rawls.model._
+
+import scala.collection.JavaConverters._
+import scala.util.{Failure, Success, Try}
+
+/**
+ * contains helper methods for working with BigQuery from an EntityProvider.
+ */
+trait DataRepoBigQuerySupport {
+
+  /**
+   * translate a single BigQuery FieldValue to a single Rawls AttributeValue
+   *
+   * @param field the BQ schema's field definition
+   * @param fv the BQ field value
+   * @return the Rawls attribute value, post-translation
+   */
+  def fieldValueToAttributeValue(field: Field, fv: FieldValue): AttributeValue = {
```

Comment:
Out of curiosity, do you know why Field uses the Legacy SQL data types instead of the Standard ones?

Reply:
I don't know, and I find it frustrating. I haven't tried to look into it, though

Reply:
You could do `field.getType.getStandardType` and match on that instead if you'd rather not use Legacy SQL

Reply:
@ansingh7115 do you prefer `field.getType.getStandardType`? It still returns a `LegacySQLTypeName` from `field.getType`, and then translates that via `getStandardType` ... so it just means we use a different enum in the `case` clauses, but doesn't change any functionality at all.